config/config.go

```diff
@@ -392,6 +389,11 @@ func (s *DestinationSubnet) GetRelayerAccountInfo() (*ecdsa.PrivateKey, common.A
 
 // GetSourceIDs returns the Subnet and Chain IDs of all subnets configured as a source
 func (c *Config) GetSourceIDs() ([]ids.ID, []ids.ID, error) {
+	// If we've already called this method, return the already constructed result
+	if len(c.sourceSubnetIDs) > 0 && len(c.sourceChainIDs) > 0 {
```

Comment:
Nit: We're introducing a couple different places where various of types of initialization occurs in this PR (in `SourceSubnet.Validate` and also here). I think it will be more maintainable to structure the initialization steps using the same patterns. Could we move this step to `Config.Validate` to match?

Reply:
Sure thing. Moved this to `Config.Validate`, and just return the stored values in this method.